### PR TITLE
Feature/preexport

### PR DESCRIPTION
**Pull request by memsharded**
This PR https://github.com/conan-io/conan/pull/3308 might be controversial, has 2 features: display & preexport method.

This current PR extracts the ``preexport()`` method, which would help cleaning some usages of global ``get_version()`` functions.

### CODE REVIEW

File: conans/client/loader.py [hunk 1]
```diff
@@ -53,7 +53,10 @@ def load_class(self, conanfile_path):
 
     def load_export(self, conanfile_path, name, version, user, channel):
         conanfile = self.load_class(conanfile_path)
-
+        conanfile = conanfile(self._output, self._runner, user, channel)
+        if hasattr(conanfile, "preexport") and callable(conanfile.preexport):
+            conanfile.recipe_folder = os.path.dirname(conanfile_path)
```
Comment (danimtb) [inline conans/client/loader.py]: not sure about this new attribute... might be confusing like `conafile_folder` was.

File: conans/test/functional/preexport_test.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+class PreExportTest(unittest.TestCase):
+
+    def load_from_file_test(self):
+        client = TestClient()
+        conanfile = """from conans import ConanFile, load
+import os
+class Pkg(ConanFile):
+    name = "MyPkg"
+
+    def preexport(self):
```
Comment (danimtb) [inline conans/test/functional/preexport_test.py]: Syntax ``preexport()`` seems almost like a typo. `pre-export()`, `pre_export()`... whichever option looks better for me. But, is this really intended for "pre export" usage only or to load information even for local commands?

Comment (memsharded) [reply conans/test/functional/preexport_test.py]: Yeah, agree, the name can be changed, maybe ``pre_export()`` is ok. ``pre-export()`` is not valid python.

File: conans/client/loader.py (hunk 1 above)
Comment (memsharded) [reply conans/client/loader.py]: Can remove it and let users use ``__file__``.

**Comment by lasote** [comment]
After today's discussion, we didn't agree on this feature. Should we close this PR?

**Pull request closed without merging**